### ISSUE TITLE
4.x: Log contract as per interface

### ISSUE THREAD

**Issue by dereuromark** [cakephp/cakephp#12971]
I would like to point out that the Psr\Log\LoggerInterface has the following contract specified:
```
    /**
     * Logs with an arbitrary level.
     *
     * @param mixed  $level
     * @param string $message
     * @param array  $context
     *
     * @return void
     */
    public function log($level, $message, array $context = array());
```

Our FileLog and others:

    public function log($level, $message, array $context = []): bool

which technically violates the void return type.

We should fix this and remove all returning values here to comply with the interface.

**Comment by saeideng** [comment]
>We should fix this and remove all returning values here to comply with the interface.

just for this reason , no good idea IMO

**Comment by dereuromark** [comment]
Why? logging either does sth or fails with exception. returning is bongers anyways.

**Comment by ADmad** [comment]
> just for this reason , no good idea IMO

If we are going to ignore contracts then there's no point implementating the interface :)

**Comment by markstory** [comment]
When we implemented the interface we were likely trying to strike a balance between full compliance and breaking existing functionality. Changing the return type now is reasonable I think.

**Comment by ADmad** [comment]
Yes it was a reasonable compromise for 3.x which needs to be now fixed in 4.0.
